### ISSUE TITLE
Interested in Swift version and framework-support

### ISSUE THREAD

**Issue by michaeldorner** [michaeldorner/BeeTee#7]
Are there any interests in a Swift version and refactoring the code into a [Xcode standalone-framework](https://developer.apple.com/library/mac/documentation/MacOSX/Conceptual/BPFrameworks/Tasks/CreatingFrameworks.html)?

**Comment by simgul-1** [comment]
Yes!

**Comment by jbarmanet** [comment]
Yeah! +1

**Comment by cansurmeli** [comment]
Definitely 👍🏻

**Comment by alanscarpa** [comment]
+1

**Comment by llKoull** [comment]
Hi!

¿Any news there? I trying to use your MDBluetooth... classes in a swift project, but when I try to use the Bluetooth private headers, i'm getting this error:

`Undefined symbols for architecture arm64:
  "_OBJC_CLASS_$_BluetoothManager", referenced from:
      objc-class-ref in BluetoothViewController.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)`

I made a bridging header that just imports your MDBluetoothHeader class (this class is internally calling BluetoothHeader, MDBluetoothDevice, ...) but I don't know if it is wrong.

`#ifndef Demo_Bridging_Header_h
#define Demo_Bridging_Header_h

#import "MDBluetoothManager.h"

#endif `

Do you know if i should import anything else?

Thanks and best regards!

**Comment by michaeldorner** [comment]
Unfortunately I don't have time at the moment for working at BeeTee; maybe while christmas.

**Comment by llKoull** [comment]
Hi @michaeldorner !

Thanks for your fast answer ;) And no one tried to use your actual code (written in Objective-C) in a Swift project using Bridging headers?

Best regards,

**Comment by michaeldorner** [comment]
I've got some feedback, that is works, even [for iOS 10.1.1](https://github.com/michaeldorner/BeeTee/issues/11#issuecomment-262469621).

**Comment by llKoull** [comment]
Yeah, I tried to compile your BeeTee project directly downloaded from your repo and It worked in my iPhone 7 with the latest iOS version. The problem is when i try to import your MDBluetooth... classes into another project that is written in swift.

**Comment by michaeldorner** [comment]
Santa Claus brought some Christmas presents: BeeTee 3 in Swift and completely refactored! Let me know what you think about it!

Merry Christmas and a happy new year!

**Comment by llKoull** [comment]
Thanks Michael! I will try It ^_^ 

Does anyone know if this can be used to send/receive data from a bluetooth device??

Merry Christmas!!! 😄😄😄

**Comment by jbarmanet** [comment]
Thank you Michael, great job. Not tried yet but eager to... after Xmas Party :)
Merry Christmas!

**Comment by michaeldorner** [comment]
@llKoull I haven't checked this so far, but connecting is still the hardest part.

**Comment by llKoull** [comment]
@michaeldorner thanks to your code, now i can search, connect and disconnect bluetooth devices 😄

But I don't even know which method can be used to transfer data, I tried trough the private frameworks, using the External Accessory framework or the GameKit framework, but i'm not able to send and/or receive information :S

It seems that the private framework does not have any method for this propose (or i did not find it).

Thanks!
